src/common/gpio_servo.rs

```diff
@@ -0,0 +1,195 @@
+//! A generalized servo implementation that uses a PWM signal on a GPIO pin
+//! via methods on a board component
+//!
+//! # Creating a GPIO pin servo and turning it 90 degrees
+//!
+//! ```ignore
+//!
+//! let board = FakeBoard::new(vec![]);
+//!
+//! let servo_settings = GpioServoSettings {
+//!     min_angle_deg: 0,
+//!     max_angle_deg: 180,
+//!     min_period_us: 500,
+//!     max_period_us: 2500,
+//!     pwm_resolution: 0
+//! }
+//!
+//! let mut servo = GpioServo::new(board, 12, servo_settings);
+//!
+//! servo.move_to(90).unwrap()
+//!
+//! ```
+
+use std::sync::{Arc, Mutex};
+
+use super::{
+    actuator::Actuator,
+    board::{Board, BoardType},
+    config::{AttributeError, ConfigType},
+    registry::{get_board_from_dependencies, ComponentRegistry, Dependency},
+    servo::{Servo, ServoType},
+    status::Status,
+};
+
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
+    if registry.register_servo("gpio", &from_config).is_err() {
+        log::error!("gpio model is already registered")
+    }
+}
+
+#[allow(dead_code)]
+pub(crate) fn from_config(
+    cfg: ConfigType,
+    dependencies: Vec<Dependency>,
+) -> anyhow::Result<ServoType> {
+    let board = get_board_from_dependencies(dependencies);
+    if board.is_none() {
+        return Err(anyhow::anyhow!(
+            "actual board is required to be passed to configure a gpio servo"
+        ));
+    }
+    let board_unwrapped = board.unwrap();
+    let servo_settings = GpioServoSettings::from_config(&cfg)?;
+    let pin = cfg.get_attribute::<i32>("pin").map_err(|err| match err {
+        AttributeError::ConversionImpossibleError => {
+            anyhow::anyhow!("could not convert pin attribute format when configuring gpio servo")
```

Comment:
@gvaradarajan - I won't insist on it, but if you feel like improving this error message before merging I think it would be nice for our users. Resolving this comment though, and no need to post an update should you chose to improve the error message.